opm/simulators/wells/WellStateFullyImplicitBlackoil.hpp

```diff
@@ -552,7 +554,8 @@ namespace Opm
 
             for( const auto& wt : this->wellMap() ) {
                 const auto w = wt.second[ 0 ];
-                if (!this->open_for_output_[w])
+                const auto& pwinfo = *parallel_well_info_[w];
+                if (!this->open_for_output_[w] || !pwinfo.isOwner())
```

Comment:
Is it correct to skip non-owned wells here? In the WellState class, for the more basic connection data you used a dummyWell to accumulate the data for non-owned wells, then communicated it. Is that not what we must do here as well?

Reply:
I think I understand it now: the code for connections that used to be below this point is actually called from the super-class already since the reportConnections() is virtual.